Fix -Werror failure due to deprecated Qt functions

ff2000 opened the pull request:
I used the hints given by the deprecation warning messages most of the times, besides qApp->desktop()->screenGeometry().
Also look at the QColorDialog::getRgba fixes: Do you really want the Alpha Channel?

ff2000 commented:
QList::swapItemsAt was introduced in Qt-5.13, immediately deprecating QList::swap. So either we decide for which Qt version -Werror fails or I add an #if to check for Qt version or I add -Wno-error=deprecated.

matevz commented:
> Also look at the QColorDialog::getRgba fixes: Do you really want the Alpha Channel?

Yeah, alpha channel should be there.

matevz commented:
> QList::swapItemsAt was introduced in Qt-5.13, immediately deprecating QList::swap. So either we decide for which Qt version -Werror fails or I add an #if to check for Qt version or I add -Wno-error=deprecated.

Please wrap this inside
```
#if QT_VERSION >= ...
```

Canorus should be compilable at least on the latest Ubuntu LTS (or better two Ubuntu LTSes) which have Qt 5.9 and Qt 5.12 respectively. If CI tests pass for you, then you're fine.

matevz commented on src/score/playablelength.cpp:
```diff
@@ -213,7 +213,7 @@ QList<CAPlayableLength> CAPlayableLength::timeLengthToPlayableLengthList(int t,
     if (!longNotesFirst)
         for (i = 0, j = pl.size() - 1; i < j; i++, j--)
-            pl.swap(i, j);
+            pl.swapItemsAt(i, j);
```
Wrap the `swapItemsAt()` inside
```
#if QT_VERSION >= ...
```
because it's not available prior to Qt 5.13.

matevz commented on src/layout/drawablemark.cpp:
```diff
@@ -58,7 +58,7 @@ CADrawableMark::CADrawableMark(CAMark* mark, CADrawableContext* dContext, double
         QFontMetrics fm(font);
 
-        int textWidth = fm.width(static_cast<CAText*>(this->mark())->text());
+        int textWidth = fm.horizontalAdvance(static_cast<CAText*>(this->mark())->text());
```
`horizontalAdvance()` was introduced in Qt 5.11, so please also wrap this one inside
```
#if QT_VERSION >= ...
```

ff2000 replied:
OK, will wrap those 7 calls in QT_VERION check.

ff2000 commented:
Now it fails at QFontMetrics::horizontalAdvance, which was introduced with Qt-5.11.
Same procedure? This isn't just one occurrence but seven.
The lowerBound/upperBound changes shouldn't hurt as it's a port to STL.

ff2000 commented:
I added two more commits:
* Remove outdated QT_VERSION checks (besides the one in helpbrowser.cpp) to clean up the sources
* adjust CMakeLists.txt for min version: 5.2.1 -> 5.5.1 (Qt version of 16.04)

I wanted to also adjust the README but I don't know how to deal with webkit vs. webengine. helpbrowser.cpp seems to be not compilable ATM. Also the README encourages to install qtwebkit AND qtwebengine.
And it says "Requires latest Qt: v5.6", also the tested versions are 5.6.2 and 5.9.5. I leave it to you to decide on that matter.

matevz commented on CMakeLists.txt:
```diff
@@ -87,8 +87,9 @@ ENDIF(APPLE)
-SET(QT_MIN_VERSION "5.2.1")
+# Search for Qt5. We need the version available in the last two Ubuntu LTS
+# Releases
+SET(QT_MIN_VERSION "5.5.1")
```
Leave the minimum version at 5.2.1. We want to be backward compatible as much as possible. I don't see a good reason why 5.5.1 should be minimum rather than that it's shipped with Ubuntu 16.04.

ff2000 replied:
I reverted the entire commit, so CMakeLists.txt stays untouched.

matevz commented on src/widgets/actionseditor.cpp:
```diff
@@ -69,14 +69,11 @@ QList <QKeySequence> ActionsEditor::stringToShortcuts(QString shortcuts) {
 	for (int n=0; n < l.count(); n++) {
 		//qDebug("%s", l[n].toUtf8().data());
-#if QT_VERSION >= 0x040300
+
```
Also remove the comment regarding Qt 4.3 and 4.4 versions below.

ff2000 replied:
Is that enough?
``` diff
-               // Qt 4.3 and 4.4 (at least on linux) seems to have a problem when using Traditional Chinese
+               // Qt (at least on linux) seems to have a problem when using Traditional Chinese
```
And is the whole workaround still needed with Qt5?

matevz replied:
@suamor Should this be cleaned up on Qt5? What "arrow keys" did `QKeySequence` delete?

matevz replied:
@ff2000: go ahead and apply this comment fix. We will open the issue later, if the error is still there.

matevz commented:
> I wanted to also adjust the README but I don't know how to deal with webkit vs. webengine. helpbrowser.cpp seems to be not compilable ATM. Also the README encourages to install qtwebkit AND qtwebengine.

Having `qtwebengine5-dev` package installed on my Ubuntu 18.04 compiles the help widget fine. Some other distributions name this package differently, so it's pretty messy. Maybe @suamor knows more about this.

In any case, this package is optional and it renders the User's guide in a side pane of the main window. If it's not there (for e.g. current raspbian doesn't ship this package), the User's guide is opened in a default web browser externally.

ff2000 commented:
OK, I was wrong regarding with the helpbrowser.
YouCompleteMe couldn't find WebEngine headers (not installed) and its internal FixIt support seemed to kinda mark the type, ingeniously with a trailing underscore... I didn't realise that and thought `QWebEngineView_` was a typedef to try to support both WebKit and WebEngine (helpbrowser.cpp has QT_VERSION check in place to achieve that) and the typedef simply got lost at some point... Sorry for that false assumption...

I can cleanup canorus from QtWebKit traces if you want, but in a seperate PR.

ff2000 commented:
@matevz I changed the comment and squashed commits. I don't get the Merge button unlocked - I'm not authorized. Clicked on the "Update branch" button which unfortunately merged master.
So could you merge this PR if you are happy with it, please? Thx!

matevz commented:
Merged. Thanks for your effort!
